### ISSUE TITLE
[MAYA-125954] Expose command/interface to indicate which variant to select on import of USD

### ISSUE THREAD

**Issue by tanant** [Autodesk/maya-usd#2685]
### Originally discussed in https://github.com/Autodesk/maya-usd/discussions/2214

Opening new issue to capture the work. On import, the UI provides a way to select the particular variants desired, and it'd be good to provide a similar capability for the file importer so that loads and imports can pick correct variant without needing to open the UI up.

Note - this is specifically for the pathway where the USD stage is imported _into_ the maya data model (i.e. translated to native maya, not the other pathway where we keep it as USD).

<div type='discussions-op-text'>

<sup>Originally posted by **jcoelho88** March 23, 2022</sup>
Hello,

I want to be able to import my USD assets with variant selection via python.

When importing with the import dialog there is a option to open a hierarchy view and select the variants on what ever prim you want.

The command pops up in the script editor when its ran but I do not see how any of the options chosen in the Hierarchy view is being applied.

Currently an import command I use could look like this.

```python
cmds.file(
            assetusd_path, i=True, type='USD Import', ignoreVersion=True,
            renameAll=True, mergeNamespacesOnClash=False, namespace=namespace,
            options=";shadingMode=[[none,default]];preferredMaterial=none;primPath=/;readAnimData=0;"
                    "useCustomFrameRange=0;startTime=0;endTime=0;importUSDZTextures=0",
            preserveReferences=True, importFrameRate=True, importTimeRange='override',
        )
```

How do I include the options for variant selection on specific prims via the python import command?

Any help would be appreciated.

Cheers,

Jason Coelho</div>

**Comment by tanant** [comment]
admin action - internally logged as MAYA-125954

**Comment by erikstantoncarlson** [comment]
To add to this, I imagine there could be a `variants` keyword arg which could be a dictionary such as
```
{
    '{prim_path}': {
        '{variant_set}': '{variant}'
    }
}
```
Example:
```
variants = {
    '/asset/geo': {
        'model': 'damaged',
        'look': 'damaged'
    },
    '/asset/hair': {
        'hair': 'wet'
    }
}
cmds.file(asset_usd_path, i=True, type='USD Import', options="variants={variants};...".format(variants), ...)
```

Not completely sure how `options` would represent a dictionary value since it's all stringified, but I hope you get the idea.

**Comment by tanant** [comment]
Would there be a major benefit using a dict over a flat repeatable tuple of (prim_path, variant_set, variant)? e.g. `[('/asset/geo', 'look', 'damaged'), ('/asset/geo', 'model', 'damaged'), ('/asset/hair', 'hair', 'wet')]`

**Comment by dolivares-spinvfx** [comment]
The tuple approach may be better than a dict to maintain order for nested variant sets (at least in python 2.7)

usd primspec paths specify variants like this:
/path/to/prim{variant_set=variant}continuing/path/to/a/child/prim{another_variant_set=variant}etc/etc

**Comment by jcoelho88** [comment]
@tanant It looks like this has been addressed in Maya-usd 0.21.0

USD import command support prim variants https://github.com/Autodesk/maya-usd/pull/2703
```
Add a new -primVariant (-pv) flag to the mayaUSDimport command.
Takes three argument: prim path, variant name, variant selection.
Can be repeated.
Extended an existing unit test to test it.
```
I'll have to test it after updating to the latest plugin.

**Comment by santosd** [comment]
Nice! Thank you for validating that @jcoelho88 I will go ahead and close this ticket out for the time being.
